lxd/db/warnings_types.go

```diff
@@ -0,0 +1,81 @@
+//go:build linux && cgo && !agent
+// +build linux,cgo,!agent
+
+package db
+
+// WarningType is a numeric code indentifying the type of warning.
+type WarningType int
+
+// Possible values for WarningType
+//
+// WARNING: The type codes are stored in the database, so this list of
+//          definitions should be normally append-only. Any other change
+//          requires a database update.
+const (
+	WarningClusterTimeSkew WarningType = iota
+	WarningMissingCgroups
+	WarningUnableToConnectToMAAS
+	WarningOfflineClusterNodes
+	WarningMissingAppArmorSupport
+	WarningIPv6NetworksWithLargerPrefixThan64
+	WarningApparmorDisabledDueToRawDnsmasq
+	WarningNetworkStartupFailure
+	WarningBrNetfilterForProxy
+	WarningMissingVirtiofsd
+)
+
+// Description return a human-readable description of the warning type.
+func (t WarningType) Description() string {
+	switch t {
+	case WarningClusterTimeSkew:
+		return "Time skew detected between leader and local"
+	case WarningMissingCgroups:
+		return "Couldn't find CGroup"
+	case WarningUnableToConnectToMAAS:
+		return "Unable to connect to MAAS"
+	case WarningOfflineClusterNodes:
+		return "Cluster node is offline"
+	case WarningMissingAppArmorSupport:
+		return "Missing AppArmor support"
+	case WarningIPv6NetworksWithLargerPrefixThan64:
+		return "IPv6 networks with a prefix larger than 64 aren't properly supported by dnsmasq"
+	case WarningApparmorDisabledDueToRawDnsmasq:
+		return "AppArmor disable due to raw.dnsmasq"
+	case WarningNetworkStartupFailure:
+		return "Network startup failure"
+	case WarningBrNetfilterForProxy:
+		return "Proxy bridge netfilter not enabled"
+	case WarningMissingVirtiofsd:
+		return "Missing virtiofsd"
+	}
+
+	return "Unspecific warning"
+}
+
+// Severity returns the severity of the warning type.
+func (t WarningType) Severity() string {
+	switch t {
+	case WarningClusterTimeSkew:
+		return "low"
```

Comment:
@monstermunchkin feels like severity should be its own Go type, along with constants for the different levels so that the return value of this function can be compared to the discrete severity levels we define?